docs/ApiGuidelines.md

```diff
@@ -0,0 +1,76 @@
+# API Guidelines
+
+## APIs
+
+There are three types of APIs: 
+
+1. Public
+2. Experimental
+3. Internal
+
+### Public APIs
+
+Public APIs are the supported APIs that apps can use. 
+
+Public API requirements: 
+
+* All public APIs must have at least one test case.
+* All public APIs must be in a header file that is published to a location under the build directory.
```

Comment:
It seems that the "build directory" term means something different here than what it does in CMake, so I think someone might get confused. In CMake, the intermediaries and results of the build go to the build directory, so headers generally won't be there. To add something to the SDK package, you would add it to the "install" directory, like here:
`install(FILES openenclave/enclave.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/openenclave/)`.

Maybe change "build directory" to something like "SDK output directory" and if you want to mention CMake specifically, you can also specify it must be under  `${CMAKE_INSTALL_INCLUDEDIR}/openenclave`.

Reply:
Yes, I agree with @gupta-ak. This may be slightly confusing as there is existing tech debt removing the manual placement of some libraries/binaries etc. in the build directory (which as Akash explained, is unnecessary as the build directory is _only_ intermediate artifacts, and should just be structured to the default locations used by CMake).

Reply:
Comment is outdated, as GitHub shows.  Current text says "SDK output directory" as suggested.